src/support/thread_group.c

```diff
@@ -22,11 +22,21 @@ __wt_thread_run(void *arg)
 	thread = (WT_THREAD*)arg;
 	session = thread->session;
 
+	for (;;) {
+		if (!F_ISSET(thread, WT_THREAD_RUN))
+			return (WT_THREAD_RET_VALUE);
+		if (!F_ISSET(thread, WT_THREAD_ACTIVE))
+			__wt_cond_wait(session, thread->pause_cond,
+			    WT_THREAD_PAUSE * WT_MILLION, thread->chk_func);
```

Comment:
Pausing for 10 seconds here makes me nervous. I know the condition variable is signaled if something more urgent is required, but we've seen cases before where having such long sleeps can lead to very slow shutdown (for example). I'd be happier if this was 10x/second - 0.1 of a second isn't really noticeable.